File: surf-api-bukkit/surf-api-bukkit-api/src/main/java/dev/slne/surf/surfapi/bukkit/api/nms/listener/packets/serverbound/SignUpdatePacket.java
package dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound;

import io.papermc.paper.math.BlockPosition;
import org.checkerframework.common.value.qual.ArrayLen;
import org.jetbrains.annotations.ApiStatus.NonExtendable;
import org.jetbrains.annotations.Range;

@NonExtendable
public non-sealed interface SignUpdatePacket extends NmsServerboundPacket {

  BlockPosition getPosition();

  @ArrayLen(4)
  String[] getLines();

  String getLine(@Range(from = 1, to = 4) int line);

  boolean isFrontText();
}

File: surf-api-paper/surf-api-paper/src/main/kotlin/dev/slne/surf/api/paper/nms/listener/packets/serverbound/SignUpdatePacket.kt
package dev.slne.surf.api.paper.nms.listener.packets.serverbound

import dev.slne.surf.api.paper.nms.NmsUseWithCaution
import io.papermc.paper.math.BlockPosition
import org.jetbrains.annotations.Range

@NmsUseWithCaution
interface SignUpdatePacket : NmsServerboundPacket {
    val position: BlockPosition
    val lines: Array<String>
    val isFrontText: Boolean

    fun getLine(line: @Range(from = 1, to = 4) Int): String
}
